x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/querydsl/query/BoolQuery.java

```diff
@@ -23,52 +29,59 @@ public class BoolQuery extends Query {
      * {@code true} for boolean {@code AND}, {@code false} for boolean {@code OR}.
      */
     private final boolean isAnd;
-    private final Query left;
-    private final Query right;
+    private final List<Query> queries;
```

Comment:
Imo, this change is a bit forced, conceptually speaking. A boolean query by definition is a boolean operation between two expressions, just like a math operation is. Having multiple expressions ALL linked together by either OR or AND seems like a specialized boolean query, MultiBoolQuery if you want. The BoolQueryBuilder in ES is built similarly, but is generalized: a list of must, a list of should, a list of filters etc.

I would have expected a solution around "flattening" the bool statements, but not touching the BoolQuery in this way.

Reply:
Confirmed with the search team that the two types of queries are equivalent as Lucene normalizes them to the same query (the one created above) when `minimum_should_query` is the default (1 - which it is).